### PR TITLE
[DO NOT MERGE] a11y fixes

### LINKED ISSUE #5375: [a11y] 9.1.1.1a - Alternative texts for control elements
- Context: The Web UI should be accessible according to the WCAG2.1 /BITV 2.0 guidelines. The number in the title relates to the test-steps: https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/pruefschritte.html
- "Status" relates to the evaluation scheme https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/verfahren.html#c1262 
- "a11y" is short for accessibility
- ----------------
## General notes
The alternative text for the logo "Zur Seite "alle Dateien" navigieren" should contain the text"Logo own cloud.“

## Status: rather fulfilled

### LINKED ISSUE #5378: [a11y] 9.1.3.1b - HTML-Structure elements for lists
- Context: The Web UI should be accessible according to the WCAG2.1 /BITV 2.0 guidelines. The number in the title relates to the test-steps: https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/pruefschritte.html
- "Status" relates to the evaluation scheme https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/verfahren.html#c1262 
- "a11y" is short for accessibility
- ----------------

## General notes
Pages 1 and 2: The tags (Einstellungsoptionen - "Betrachter", etc.) in the area "Öffentliche Links" are visually a list which is not marked up with list elements.
The tags in the area "Personen" are correctly implemented.

## Status: partly fulfilled

### LINKED ISSUE #5379: [a11y] 9.1.3.1d - Structured content
- Context: The Web UI should be accessible according to the WCAG2.1 /BITV 2.0 guidelines. The number in the title relates to the test-steps: https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/pruefschritte.html
- "Status" relates to the evaluation scheme https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/verfahren.html#c1262 
- "a11y" is short for accessibility
- ----------------

Page 3 (Gelöschte Dateien)
The sentence "vor ein paar Sekunden" should be implemented with a structure element for paragraphs ( p ).

## Status: rather fulfilled

### LINKED ISSUE #5381: [a11y] 9.1.3.1g - No structure markup for layout tables
- Context: The Web UI should be accessible according to the WCAG2.1 /BITV 2.0 guidelines. The number in the title relates to the test-steps: https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/pruefschritte.html
- "Status" relates to the evaluation scheme https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/verfahren.html#c1262 
- "a11y" is short for accessibility
- ----------------

Only note: Layout table "Versionen": For layout tables the ARIA role="presentation" should be used.

**relates:**
- Page 1 (Alle Dateien)
- Page 2 (Mit anderen geteilten Dateien)

## Status: fulfilled

### LINKED ISSUE #5383: [a11y] 9.1.3.2 - Logical order
- Context: The Web UI should be accessible according to the WCAG2.1 /BITV 2.0 guidelines. The number in the title relates to the test-steps: https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/pruefschritte.html
- "Status" relates to the evaluation scheme https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/verfahren.html#c1262 
- "a11y" is short for accessibility
- ----------------

## General notes
- **Pages 1 to 3:** Above the h1-heading is a visually hidden text which tells the users if the elements in the table are selected or not, e.g. "Keine Elemente ausgewählt.". This text should be placed after the h1-heading, otherwise it could be missed out by users when navigating via headings.

## Status: rather fulfilled

### LINKED ISSUE #5395: [a11y] 9.3.2.4 - Consistent labelling
- Context: The Web UI should be accessible according to the WCAG2.1 /BITV 2.0 guidelines. The number in the title relates to the test-steps: https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/pruefschritte.html
- "Status" relates to the evaluation scheme https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/verfahren.html#c1262 
- "a11y" is short for accessibility
- ----------------
## General notes
Responsive (Mobile) view: when the menue (Hamburger-Button) is closed the label of the button is "Navigationsmenü  ffnen" and when the user wants to close the menue the label of the button is "Seitenleiste schließen".
This could lead to irritations for the users. Also the area on the right of the page "Seitenleiste" appears as label in the close button "X".

## Status: partly fulfilled

### LINKED ISSUE #5396: [a11y] 9.3.3.1 - Error detection
- Context: The Web UI should be accessible according to the WCAG2.1 /BITV 2.0 guidelines. The number in the title relates to the test-steps: https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/pruefschritte.html
- "Status" relates to the evaluation scheme https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/verfahren.html#c1262 
- "a11y" is short for accessibility
- ----------------
## Status: partly fulfilled
- **Page 1 (Alle Dateien)**
When creating a public link and in case the user leaves the fields empty, an error message appears above the form:
**a)** The error message is in English language but should be in German - "error creating public share"
**b)** The text of the error message is not informative because it doesn’t explain what went wrong (what data caused the error?)
**c)** The error message is not handled by screenreaders. One solution could be the use of aria- live or role="alert“.

## Status: fulfilled
- **Page 2 (Mit anderen geteilten Dateien)**

## Status: not applicable
- **Page 3 (Gelöschte Dateien)**
- **Page 4 (Konto)**

### LINKED ISSUE #5397: [a11y] 9.4.1.2 - Name, role, value available
- Context: The Web UI should be accessible according to the WCAG2.1 /BITV 2.0 guidelines. The number in the title relates to the test-steps: https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/pruefschritte.html
- "Status" relates to the evaluation scheme https://www.bitvtest.de/bitv_test/das_testverfahren_im_detail/verfahren.html#c1262 
- "a11y" is short for accessibility
- ----------------
## General notes
- **All pages:** for the elements of the user menue (Liste mit "Profil", "Einstellungen", etc.) ARIA role="tooltip" was used. In this case this is not correct as it has an other purpose - "A contextual popup that displays a description for an element.". The roles should be removed.
Also using the aria-describedby attribute on the button "Benutzermenü" in the expanded sub menue is not necessary.
- **Pages 1 to 3 / modal dialogues:** using aria-labelledby ( aria-labelledby="ocmodal-title" ) is not allowed on div und span elements as screenreader won’t handle the attribute reliable. An accessible sample can be found here: https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/dialog.html.
- **Note:** the anchor links (in this case "Zum Hauptinhalt springen") should be implemented as links and not as buttons.

## Status: rather fulfilled
- **Page 4 (Konto)**
See general notes

## Status: partly fulfilled
- **Page 1 (Alle Dateien)**
When copying or moving an element (Folder or file) a new page is opened.
On opening (because of using aria-live="polite“) the text "Zur Seite files- location-picker navigiert" is read by screenreaders. This text contains an English variable and is not informative. A better solution would be to remove aria-live="polite" and move the keyboard focus with tabindex="-1" on the h1-heading on page load.
See general notes

- **Page 2 (Mit anderen geteilten Dateien)**
See general notes

- **Page 3 (Gelöschte Dateien)**
See general notes

### PR DESCRIPTION
**Pull request by pascalwengerter**
DO NOT MERGE - WIP PR

## Description
Implementing feedback from the a11y audit

## Related Issue
- Fixes #5375
- Fixes #5378
- Fixes #5379
- Fixes #5381
- Fixes #5383
- Partly fixes #5397 (page title for location picker)
- Fixes #5396
- Fixes #5395

Also, https://github.com/owncloud/ocis/pull/2309 fixes two more issues (#5376 and #5377)

## Open tasks:
- Should fix some more issues between # 5375 and # 5398

### CODE REVIEW

File: packages/web-runtime/src/App.vue [hunk 1]
```diff
@@ -182,7 +182,7 @@ export default {
     },
 
     sidebarLogoAlt() {
-      return this.$gettext('Navigate to all files page')
+      return this.$gettext('Logo ownCloud, navigate to all files page')
```
Comment (kulmann) [inline packages/web-runtime/src/App.vue:185]: TODO: use general.name from theme instead of hardcoded `ownCloud`. Needs to use `$gettextInterpolate` then.

Comment (pascalwengerter) [reply packages/web-runtime/src/App.vue]: Addressed in the most recent commit, great idea!

**Comment by JammingBen** [comment]
A lot of merge conflicts as the code base changed heavily over the last few months. @pascalwengerter what do you think, does it make sense to keep this open?

**Comment by pascalwengerter** [comment]
I'll close here since the codebase has moved on (as in "improved") _a lot_ in the last 2,5 years, and a11y has gotten its way into many new or refactored components. While the project most likely would need another audit before picking up certification, this PR can still be found if need arises, but isn't actionable enough for now.

**Pull request closed without merging**